File: lib/src/main/java/com/tencent/cubershi/plugin_loader/blocs/ParsePluginApkBloc.java
package com.tencent.cubershi.plugin_loader.blocs;

import com.tencent.cubershi.plugin_loader.exceptions.ParsePluginApkException;
import com.tencent.cubershi.plugin_loader.infos.ApkInfo;

import java.io.File;

/**
 * 解析插件apk逻辑
 *
 * @author cubershi
 */
public class ParsePluginApkBloc {
    /**
     * 解析插件apk
     *
     * @param pluginFile 插件apk文件
     * @return 解析信息
     * @throws ParsePluginApkException 解析失败时抛出
     */
    public static ApkInfo parse(File pluginFile) throws ParsePluginApkException {
        if (pluginFile.exists() && pluginFile.length() > 0) {
            return new ApkInfo();
        } else {
            throw new ParsePluginApkException("测试代码,但文件不合法");
        }

    }
}

File: plugin-loader/src/main/kotlin/com/tencent/cubershi/plugin_loader/blocs/ParsePluginApkBloc.kt
package com.tencent.cubershi.plugin_loader.blocs

import android.content.Context
import android.content.pm.PackageManager.GET_ACTIVITIES
import android.content.pm.PackageManager.GET_META_DATA
import com.tencent.cubershi.plugin_loader.exceptions.ParsePluginApkException
import com.tencent.cubershi.plugin_loader.infos.PluginActivityInfo
import com.tencent.cubershi.plugin_loader.infos.PluginInfo

/**
 * 解析插件apk逻辑
 *
 * @author cubershi
 */
object ParsePluginApkBloc {
    /**
     * 解析插件apk
     *
     * @param pluginFile 插件apk文件
     * @return 解析信息
     * @throws ParsePluginApkException 解析失败时抛出
     */
    @Throws(ParsePluginApkException::class)
    fun parse(archiveFilePath: String, hostAppContext: Context): PluginInfo {
        val packageManager = hostAppContext.packageManager
        val packageArchiveInfo = packageManager.getPackageArchiveInfo(
                archiveFilePath,
                GET_ACTIVITIES or GET_META_DATA
        )
        val pluginInfo = PluginInfo(
                packageArchiveInfo.applicationInfo.packageName
                , packageArchiveInfo.applicationInfo.className
        )
        packageArchiveInfo.activities.forEach {
            pluginInfo.putActivityInfo(PluginActivityInfo(it.name, it.themeResource))
        }
        pluginInfo.metaData = packageArchiveInfo.applicationInfo.metaData
        return pluginInfo
    }
}
